packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -46,13 +54,24 @@ export function createTokenIntrospectionMiddleware({
       const tokenIntrospectionClient = await ctx.container.use(
         'tokenIntrospectionClient'
       )
-      const tokenInfo = await tokenIntrospectionClient.introspect(token)
-      if (!tokenInfo) {
+      let tokenInfo: TokenInfo
+      try {
+        tokenInfo = await tokenIntrospectionClient.introspect(token)
+      } catch (err) {
+        ctx.throw(401, 'Invalid Token')
+      }
+      if (!tokenInfo.active) {
         ctx.throw(401, 'Invalid Token')
       }
+      ctx.clientKey = tokenInfo.key.jwk
+
+      if (!tokenInfo.access) {
+        await next()
+        return
+      }
```

Comment:
```suggestion
      if (!isActiveTokenInfo(tokenInfo)) {
        ctx.throw(401, 'Invalid Token')
      }

```

`ctx.clientKey = tokenInfo.key.jwk` can stay in its previous spot below